pkg/srv/mstream/parse_event.go

```diff
@@ -0,0 +1,191 @@
+package mstream
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"jinr.ru/greenlab/go-adc/pkg/log"
+)
+
+const wordSize = 4
+const mstreamTimeHeaderSize = 8
+
+type MstreamHeader struct {
+	EventTimestamp    uint32
+	Subtype           uint8
+	Length            uint16
+	Tainsec           uint32
+	Taiflags          uint32
+	MStreamDataHeader []MstreamDataHeader
+}
+
+type MstreamDataHeader struct {
+	DataType uint8
+	Channel  uint8
+	Spec     uint8
+	Length   uint16
+	TdcData  []interface{}
+	ADCData  AdcData
+}
+
+type TdcData2 struct {
+	Timestamp uint16
+	Evnum     uint16
+	Id        uint8
+	N         uint8
+}
+
+type TdcData3 struct {
+	Wcount uint16
+	Evnum  uint16
+	Id     uint8
+	N      uint8
+}
+
+type TdcData4 struct {
+	Rcdata uint32
+	Ledge  uint16
+	Ch     uint8
+	N      uint8
+}
+
+type TdcData5 struct {
+	Rcdata uint32
+	Tedge  uint16
+	Ch     uint8
+	N      uint8
+}
+
+type TdcData6 struct {
+	Err uint16
+	Id  uint16
+	N   uint8
+}
+
+type AdcData struct {
+	Timestamp uint16
+	Length    uint16
+	Voltage   []uint16
+}
+
+// Encode []byte to struct Mstrean signal doc from TQDC2_Data_Format_rev0.pdf and tqdc16vs.py
+func MstreamHeaderConstructor(d []byte) MstreamHeader {
```

Comment:
It is a convention to name constructor methods as NewSomthing

Reply:
I am fixed it